mantidproject/mantid
# Implement alias deprecation in Python

Companion to PR #32525  

**Description of work.**
- [x] New method `const std::string IAlgorithm.aliasDeprecated(const std::string) const;`
- [x] Expose `IAlgorithm.aliasDeprecated` to python so that it can be overriden
- [x] ConfigService option to raise upon use of a deprecated Alias
- [x] implement Alias warnings and optional raise in the algorithm wrapper
- [x] a class decorator `deprecated_alias(date)` will hide implementation of method `aliasDeprecated`
- [x] developer documentation on how to rename an algorithm

**To test:**

Run the following script in the workbench:
```python
from mantid.kernel import ConfigService
config = ConfigService.Instance()

config['algorithms.alias.deprecated'] = 'Log'
from mantid.simpleapi import CNCSSuggestTIB, SuggestTibCNCS
x = CNCSSuggestTIB(42)  # use algorithm name
x = SuggestTibCNCS(42)  # use alias
config['algorithms.alias.deprecated'] = 'Raise'
x = SuggestTibCNCS(42)  # use alias, raise RuntimeError


config['algorithms.alias.deprecated'] = 'Log'
from mantid.simpleapi import HYSPECSuggestTIB, SuggestTibHYSPEC
x = HYSPECSuggestTIB(42)  # use algorithm name
x = SuggestTibHYSPEC(42)  # use alias
config['algorithms.alias.deprecated'] = 'Raise'
x = SuggestTibHYSPEC(42)  # use alias, raise RuntimeError
```

There is no associated issue (see [PL100](https://code.ornl.gov/sns-hfir-scse/planning/-/issues/100))

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
